packages/core/src/Keymap.ts

```diff
@@ -0,0 +1,101 @@
+    pattern: ShortcutPattern;
+    priority: number;
+}
+
+// todo: questions
+// - who preventDefault / stopPropagation / stopImmediatePropagation?
+// - what about when a pattern is already present in the system?
+//   - if the check (static or dynamic) is different?
+export class Keymap {
+    _shortcutItems: ShortcutItem[] = [];
+
+    //--------------------------------------------------------------------------
+    // public
+    //--------------------------------------------------------------------------
+
+    addShortuct(
+        patternString: string,
+        commandId: CommandIdentifier,
+        commandArgs?: CommandArgs,
+        priority = 0,
+    ): void {
+        this._shortcutItems.push({
+            pattern: this._getPatternFromString(patternString),
+            commandId,
+            commandArgs,
+            priority,
+        });
+    }
+
+    getCommandCalls(keyEvent: KeyboardEvent): CommandCall[] {
+        const possibleMappings = this._shortcutItems.filter((listener): boolean => {
+            const key = listener.pattern.key;
+            const modifiers = listener.pattern.modifiers;
+
+            if (key === '*') {
+                return true;
+            }
+
+            const modifierOk =
+                ((modifiers.has('ctrl') && keyEvent.ctrlKey) ||
+                    (!modifiers.has('ctrl') && !keyEvent.ctrlKey)) &&
+                ((modifiers.has('shift') && keyEvent.shiftKey) ||
+                    (!modifiers.has('shift') && !keyEvent.shiftKey)) &&
+                ((modifiers.has('cmd') && keyEvent.metaKey) ||
+                    (!modifiers.has('meta') && !keyEvent.metaKey)) &&
+                ((modifiers.has('alt') && keyEvent.altKey) ||
+                    (!modifiers.has('alt') && !keyEvent.altKey));
+
+            if (!modifierOk) {
+                return false;
+            }
+            const keyCodeMatch = key.match(/^<(\w+)>$/);
+            const keyCode = keyCodeMatch && keyCodeMatch[1];
+            if (keyCode) {
+                return keyCode === keyEvent.code;
+            } else {
+                return key === keyEvent.key;
+            }
+        });
+        const maxPriority = Math.max.apply(null, possibleMappings.map(x => x.priority));
+        const prioritaryMapping = possibleMappings.filter(x => x.priority === maxPriority);
+        return prioritaryMapping.map(mapping => {
+            return {
+                commandId: mapping.commandId,
+                commandArgs: mapping.commandArgs,
+            };
+        });
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    _getPatternFromString(pattern: string): ShortcutPattern {
+        const patternArray = pattern
+            .trim()
+            .replace(/\s+/g, ' ')
+            .split(' ');
+        const key = patternArray.slice(-1)[0];
```

Comment:
We might want to discuss the form of the shortcut. I remember @Zinston and I were not agreeing perfectly on the capitalizaion, but we both used a `+` rather than space.

Reply:
I see any separator as equals. The one you decide will be the one used.

Reply:
We had eventually agreed on `Ctrl+a` form.

Reply:
let's uppercase